### PR TITLE
fix(backend): trim Ayrshare profile keys in request headers

### LINKED ISSUE #13101: BlockUnknownError: Illegal header value b' a83749cd-ed79-40b9-87df-2cf9e276193d'

### PR DESCRIPTION
**Pull request by pony-maggie**
- trim Ayrshare profile keys before putting them in JWT payloads and Profile-Key headers
- build request headers from a copy so per-profile headers do not mutate the client default headers
- add focused Ayrshare client tests for whitespace handling and header isolation

Fixes #13101.

## Tests
- `PYTHONPATH=. uv run --no-project --with pytest --with pytest-asyncio --with python-dotenv --with pydantic --with pydantic-settings pytest --confcutdir=backend/integrations backend/integrations/ayrshare_test.py -q`
- `PYTHONPATH=. uv run --no-project --with python-dotenv --with pydantic --with pydantic-settings python -m py_compile backend/integrations/ayrshare.py backend/integrations/ayrshare_test.py`
- `uv run --no-project --with ruff ruff check backend/integrations/ayrshare.py backend/integrations/ayrshare_test.py`
- `git diff --check`

### CODE REVIEW

**Comment by pony-maggie** [comment]
Rebased onto latest `dev` (conflict-free). CI should re-run on the new head.

**Comment by Pwuts** [comment]
🤖 Pushed a `dev` merge to this branch (758f20991e) so you don't have to chase the conflict — #14492 landed today and removed a log line two lines below your `create_post` header hunk.

Git auto-merged it, and I checked the result rather than trusting it: `dict(self.headers)` and the strip are both intact, the removed line stays removed, and `git diff origin/dev...HEAD` still comes out as exactly your +100/−2 across the same two files, so nothing of yours was lost.

Your new `backend/integrations/ayrshare_test.py` didn't collide with anything — `dev`'s Ayrshare tests live at `backend/integrations/managed_providers/ayrshare_test.py`, which is a different module.

Green on the merged head: your 2 tests plus `dev`'s 13 in `managed_providers/ayrshare_test.py` (15 passed), `block_cost_config_test.py` (23), `architecture_test.py` (3), and `blocks/test/test_block.py` (1650 passed, 84 skipped). Review to follow.

File: autogpt_platform/backend/backend/integrations/ayrshare.py [hunk 1]
```diff
@@ -74,6 +74,10 @@ def _extract_error_message(response: Any) -> str:
     return "Unknown error"
 
 
+def _normalize_profile_key(profile_key: str) -> str:
+    return profile_key.strip()
```
[the remark this answers is not in the document]
Comment (Pwuts) [reply autogpt_platform/backend/backend/integrations/ayrshare.py:78]: 🤖 Checked this one and I'm declining it — the observation is right, the proposed fix would make things worse.

Both factual halves confirmed by running them: with `profile_key="   "`, `generate_jwt` sends `profileKey=""` and `Profile-Key: ""`, while `create_post` omits the header. So the two methods do disagree on a whitespace-only key.

The fix is the problem. Making `_normalize_profile_key` raise `ValueError` changes `create_post` from omitting the header to raising for both `""` and `"   "`, and nothing catches it — the fourteen `post_to_*` blocks call `create_post` with no `try`, and `ValueError` isn't `AyrshareAPIException`, so it leaves `run()` unhandled and the executor surfaces it as `BlockUnknownError`. That is the same failure class as #13101, which is the issue this PR is fixing.

Leaving the behaviour as it stands. An empty `profileKey` reaching Ayrshare comes back as an API error through the normal `AyrshareAPIException` path, which beats an unhandled crash, and a whitespace-only stored key isn't the reported failure anyway — that one is a valid UUID with a leading space. If we do want the two methods aligned later it wants an explicit early return, not a raise.

File: autogpt_platform/backend/backend/integrations/ayrshare.py [hunk 2]
```diff
@@ -512,7 +517,9 @@ async def create_post(
         if notes:
             payload["notes"] = notes
 
-        headers = self.headers
+        headers = dict(self.headers)
+        if profile_key:
+            profile_key = _normalize_profile_key(profile_key)
         if profile_key:
             headers["Profile-Key"] = profile_key
```
Comment (Pwuts) [inline autogpt_platform/backend/backend/integrations/ayrshare.py:522]: 🤖 🔵 Nit, non-blocking. The second `if profile_key:` isn't redundant — a whitespace-only key passes the first test, strips to `""`, and the second then correctly omits the header instead of sending an empty one. But as two identical conditions in a row it reads like a slip, and the tempting cleanup (folding them into `if profile_key: headers["Profile-Key"] = _normalize_profile_key(profile_key)`) would send `Profile-Key: ""` — the same malformed-header class this PR fixes.

The walrus form keeps the behaviour and says the intent once:

```suggestion
        headers = dict(self.headers)
        if profile_key := _normalize_profile_key(profile_key or ""):
            headers["Profile-Key"] = profile_key
```

I ran this before suggesting it: both tests still pass, pyright is clean on the file, and `None` / `""` / `"   "` all still omit the header. Fine to leave it as is if you prefer the explicit form.

**Review by Pwuts — COMMENTED** [review]
## 🤖 Review — looks good to me

The Ayrshare client adds a per-user `Profile-Key` header to each outbound call. This PR fixes two bugs in that step: the header dict was bound rather than copied, so writing the profile key into it mutated the client's own defaults; and the key was used verbatim, so a stored key carrying stray whitespace went to Ayrshare as-is — the input behind #13101.

The mutation bug is a bit worse than it looks, which is worth saying because it argues for the fix. `self.headers` is the same object handed to the HTTP client as `extra_headers`, and `backend/util/request.py:573-575` applies `extra_headers` over every request's own headers. So the old write installed the profile key into the client's *default* headers, from where it would ride along on later calls from that client — including `create_profile`, which should carry no profile key at all. It is latent rather than live today, because every construction site builds a client, makes one call and drops it, but the `dict()` removes the trap before anyone caches or reuses one.

One note on #13101 so the fix isn't credited with more than it does: on the current aiohttp client that leading-space value no longer raises — I sent it through `Requests` and it reached the transport — so today's symptom is a request carrying a key Ayrshare won't match, rather than the illegal-header crash in that report. That message is h11's, which is httpx's HTTP layer, and there's no httpx anywhere in the Ayrshare path — so it's worth confirming #13101 really goes away with this rather than letting `Fixes #13101` auto-close it on merge. None of that argues against the change: the strip is the right fix, and the right layer for it: the bad value is already sitting in stored credentials, so fixing it at the write path would need a backfill, while this repairs existing rows on read.

Verified on the merged head 758f20991e: `integrations/ayrshare_test.py` + `integrations/managed_providers/ayrshare_test.py` (15 passed), `data/block_cost_config_test.py` (23), `util/architecture_test.py` (3), `blocks/test/test_block.py` (1650 passed, 84 skipped). I also reverted each fix at each site in turn — 4 mutations, 4 killed, none survived — so the new tests do fail when the code they cover regresses.

No blockers and no should-fixes; one nit inline. I triaged CodeRabbit's 🟡 on `_normalize_profile_key` and declined it on the thread — its observation is right, its proposed fix would turn a benign header omission into an uncaught `ValueError` across all fourteen `post_to_*` blocks. Auto-merge is not armed here, so an approval will not merge it on its own.

<details>
<summary>Executed tests and the mutation table</summary>

All on 758f20991e:

| suite | result |
|---|---|
| `integrations/ayrshare_test.py` + `integrations/managed_providers/ayrshare_test.py` | 15 passed |
| `data/block_cost_config_test.py` | 23 passed |
| `util/architecture_test.py` | 3 passed |
| `blocks/test/test_block.py` | 1650 passed, 84 skipped |

| # | mutation | killed by |
|---|---|---|
| 1 | `generate_jwt`: `dict(self.headers)` → `self.headers` | `test_generate_jwt_strips_profile_key_without_mutating_base_headers` |
| 2 | `generate_jwt`: drop the `_normalize_profile_key` call | same |
| 3 | `create_post`: `dict(self.headers)` → `self.headers` | `test_create_post_strips_profile_key_without_mutating_base_headers` |
| 4 | `create_post`: drop the `_normalize_profile_key` call | same |

Each mutation failed exactly the test for the method it touched and left the other green.

One gap, for the record: both tests inject `custom_requests`, so `extra_headers` is never populated and the `req_headers.update(self.extra_headers)` merge in `request.py:575` is not exercised — they prove the mutation does not happen, not that the header still arrives afterwards. I checked that half by hand: it survives, because the default dict no longer carries a `Profile-Key` to overwrite it with.
</details>

**Review by Pwuts — APPROVED** [review]

**Comment by Pwuts** [comment]
Thank you Maggie!
